delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala

```diff
@@ -0,0 +1,301 @@
+
+  /**
+    * Fetches a resolver by tag.
+    *
+    * @param id        the resolver identifier
+    * @param project   the project where the resolver belongs
+    * @param tag       the tag revision
+    */
+  def fetchBy(
+      id: Iri,
+      project: ProjectRef,
+      tag: Label
+  ): IO[ResolverRejection, Option[ResolverResource]]
+
+  /**
+    * A non terminating stream of events for resolvers. After emitting all known events it sleeps until new events
+    * are recorded.
+    *
+    * @param offset the last seen event offset; it will not be emitted by the stream
+    */
+  def events(offset: Offset = NoOffset): Stream[Task, Envelope[ResolverEvent]]
+
+}
+
+object Resolvers {
+
+  /**
+    * The resolvers module type.
+    */
+  final val moduleType: String = "resolver"
+
+  import ch.epfl.bluebrain.nexus.delta.sdk.utils.IOUtils.instant
+
+  private[delta] def next(state: ResolverState, event: ResolverEvent): ResolverState = {
+    (state, event) match {
+      // Creating a resolver
+      case (Initial, e: ResolverCreated)                            =>
+        Current(
+          id = e.id,
+          project = e.project,
+          `type` = e.`type`,
+          priority = e.priority,
+          crossProjectSetup = e.crossProjectSetup,
+          tags = Map.empty,
+          rev = e.rev,
+          deprecated = false,
+          createdAt = e.instant,
+          createdBy = e.subject,
+          updatedAt = e.instant,
+          updatedBy = e.subject
+        )
+      // Updating a resolver
+      case (c: Current, e: ResolverUpdated) if c.`type` == e.`type` =>
+        c.copy(
+          priority = e.priority,
+          crossProjectSetup = e.crossProjectSetup,
+          rev = e.rev,
+          updatedAt = e.instant,
+          updatedBy = e.subject
+        )
+
+      // Tagging a resolver
+      case (c: Current, e: ResolverTagAdded)                        =>
+        c.copy(rev = e.rev, tags = c.tags + (e.tag -> e.targetRev), updatedAt = e.instant, updatedBy = e.subject)
+
+      // Deprecating a resolver
+      case (c: Current, e: ResolverDeprecated)                      =>
+        c.copy(rev = e.rev, deprecated = true, updatedAt = e.instant, updatedBy = e.subject)
+
+      // Cases that should happen and that should have been filtered out by the evaluation
+      case _                                                        => state
+    }
+  }
+
+  private[delta] def evaluate(state: ResolverState, command: ResolverCommand)(implicit
+      caller: Caller,
+      clock: Clock[UIO]
+  ): IO[ResolverRejection, ResolverEvent] = {
+
+    def validateCrossProjectSetup(
+        resolverType: ResolverType,
+        crossProjectSetup: Option[CrossProjectSetup]
+    ): IO[ResolverRejection, Unit] = {
+      resolverType match {
+        case InProject    => IO.unit
+        case CrossProject =>
+          crossProjectSetup
+            .map { setup =>
+              if (setup.identities.isEmpty)
```

Comment:
It is fine if we don't want to allow this yet to fix the issue I described, but can you then please add a `TODO`:

```scala
//TODO: In order to fix the issue https://github.com/BlueBrain/nexus/issues/1661 we should allow empty identities
```